src/main/java/com/slimebot/games/Game.java

```diff
@@ -0,0 +1,189 @@
+package com.slimebot.games;
+
+import com.slimebot.main.Main;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.channel.Channel;
+import net.dv8tion.jda.api.entities.channel.concrete.ThreadChannel;
+import net.dv8tion.jda.api.entities.channel.middleman.MessageChannel;
+import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.dv8tion.jda.api.interactions.components.buttons.Button;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.MessageCreateAction;
+import net.dv8tion.jda.api.requests.restaction.ThreadChannelAction;
+import net.dv8tion.jda.api.utils.TimeFormat;
+
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public abstract class Game <T extends GamePlayer> extends ListenerAdapter {
+    public final UUID uuid = UUID.randomUUID();
+    public long channelId;
+    public final long guildId;
+    public final boolean playersCanJoin;
+    public short minPlayers, maxPlayers;
+    public long gameMaster;
+    public List<T> players = new ArrayList<>();
+    private final Function<Long, T> gamePlayerFunction;
+    public GameStatus status;
+
+    public Game(long gameMaster, long guildId, boolean playersCanJoin, Function<Long, T> gamePlayerFunction) {
+        this.gameMaster = gameMaster;
+        this.guildId = guildId;
+        this.playersCanJoin = playersCanJoin;
+        this.gamePlayerFunction = gamePlayerFunction;
+
+        players.add(this.gamePlayerFunction.apply(gameMaster));
+
+        PlayerGameState.setGameState(gameMaster, new PlayerGameState(this));
+
+        status = GameStatus.WAITING;
+
+        Main.jdaInstance.addEventListener(this);
+
+        // End the game if its 15min WAITING
+        Main.executor.schedule(() -> {
+            if (status == GameStatus.WAITING) end();
+        }, 15, TimeUnit.MINUTES);
+    }
+
+    public void sendJoinMessageInThread(ThreadChannelAction action) {
+        action.flatMap(channel -> {
+            this.channelId = channel.getIdLong();
+
+            return sendMessageEmbeds(buildJoinEmbed()
```

Comment:
Create a new Method `MessageEditData` buildMessage() that uses the `buildEmbed()` method and adds the additinal field and the actions rows. Use the new method for editing and use `MessageCreateData.fromEditData` for initial sending